README.md

```diff
@@ -61,8 +61,20 @@ process can safely run code defined externally.
 
 ## Application Binary Interface (ABI)
 
-The HTTP middleware ABI is not yet defined, but will be after practice. Follow
-this repository for updates.
+The HTTP middleware ABI is currently being defined. Follow this repository for updates.
+
+### Common notes
+
+- Parameters of type `string` expand to two parameters of type `u32`, the first being
```

Comment:
This has always been my biggest issue with witx, the ABI itself gets obfuscated through types like `string`. I did stick with it instead of `u32, u32` since it still does allow taking advantage of codegen and I suppose if the MD generation expanded into two params, it would be OK so can try to fix the generator for that later.